crates/blockchain-tree/src/block_indices.rs

```diff
@@ -224,12 +224,14 @@ impl BlockIndices {
     /// Remove chain from indices and return dependent chains that need to be removed.
     /// Does the cleaning of the tree and removing blocks from the chain.
     pub fn remove_chain(&mut self, chain: &Chain) -> BTreeSet<BlockChainId> {
-        let mut lose_chains = BTreeSet::new();
-        for (block_number, block) in chain.blocks().iter() {
-            let block_hash = block.hash();
-            lose_chains.extend(self.remove_block(*block_number, block_hash))
-        }
-        lose_chains
+        chain
+            .blocks()
+            .iter()
+            .flat_map(|(block_number, block)| {
```

Comment:
[`filter_map` > `flat_map` for Option](https://rust-lang.github.io/rust-clippy/master/#/filter_map)

Reply:
@DaniPopes Are we sure in this specific case because as we need to produce a `BTreeSet` at the end, this is not possible to go from `Iterator` to `BTreeSet` due to:

```
`BTreeSet<BlockChainId>` cannot be built from `std::iter::Iterator<Item=BTreeSet<BlockChainId>>`
```

so that we need to do an additional flatten step, if I'm not wrong, this would lead to:

```
pub fn remove_chain(&mut self, chain: &Chain) -> BTreeSet<BlockChainId> {
        chain
            .blocks()
            .iter()
            .filter_map(|(block_number, block)| {
                let block_hash = block.hash();
                Some(self.remove_block(*block_number, block_hash))
            })
            .flatten()
            .collect()
    }
```

Reply:
Oh sorry I thought remove_block returned an option like in the diff above